### PR TITLE
Subscribed and receiving reorg notifications and the data

### PR DESCRIPTION
**Pull request by shreekarashastry**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/spruce-solutions/quai-manager
 
 go 1.16
 
+replace github.com/spruce-solutions/go-quai => /Users/shreekara/Desktop/quai/go-quai
```
Comment (alanorwick) [inline go.mod:5]: This needs to be removed.

File: manager/main.go [hunk 2]
```diff
@@ -303,6 +306,158 @@ func (m *Manager) subscribePendingHeader(sliceIndex int) {
...
+// subscribeReOrg subscribes to the reOrg events so that we can send the reorg
+// information to clients in lower contexts
+func (m *Manager) subscribeReOrg() {
+
+	prime := "prime"
+	regions := [3]string{"region1", "region2", "region3"}
+
+	// subscribe to the prime and region clients
+	m.subscribeReOrgClients(m.miningClients[0], m.miningAvailable[0], prime, 0)
+	m.subscribeReOrgClients(m.miningClients[1], m.miningAvailable[1], regions[m.location[0]-1], 1)
```
Comment (alanorwick) [inline manager/main.go:319]: why are we initializing regions when we can use m.location?

File: manager/main.go [hunk 3]
```diff
@@ -303,6 +306,158 @@ func (m *Manager) subscribePendingHeader(sliceIndex int) {
...
+				if i != regionLoc || j != zoneLoc {
+					if m.availableClients[i].zonesAvailable[j] {
+						m.availableClients[i].zoneClients[j].SendReOrgData(context.Background(), header)
+					}
+				}
+			}
+		}
+	} else {
+		// send to only the respective zones
+		reorgLocation := getRegionIndex(location)
```
Comment (alanorwick) [inline manager/main.go:422]: We wont need getRegionIndex if we make location the m.location.

File: manager/main.go [hunk 4]
```diff
@@ -303,6 +306,158 @@ func (m *Manager) subscribePendingHeader(sliceIndex int) {
...
+		if newChain[i].Nonce == (types.BlockNonce{}) {
+			return false
+		}
+	}
+	return true
+}
+
+// compareDifficulty compares 2 chains and returns true if the newChain is heavier
+// than the oldChain and false otherwise
+func compareDifficulty(commonHead *types.Header, oldChain, newChain []*types.Header, difficultyContext int) bool {
```
Comment (alanorwick) [inline manager/main.go:350]: this method name should be more descriptive.

**Review by alanorwick — CHANGES_REQUESTED** [review]

File: manager/main.go (hunk 3 above)
Comment (shreekarashastry) [reply manager/main.go:422]: This is not the same as m.location

We are subscribing to prime and 3 regions, this location references this

File: manager/main.go (hunk 2 above)
Comment (shreekarashastry) [reply manager/main.go:319]: cannot use as explained below

File: go.mod (hunk 1 above)
Comment (shreekarashastry) [reply go.mod:5]: removed

File: manager/main.go (hunk 4 above)
Comment (shreekarashastry) [reply manager/main.go:350]: suggest a better name for comparing difficulty

Comment (alanorwick) [reply manager/main.go:350]: compareReorgDifficulty

Comment (shreekarashastry) [reply manager/main.go]: thanks, changed

**Review by alanorwick — APPROVED** [review]

**Pull request merged**